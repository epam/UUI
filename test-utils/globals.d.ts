declare module "jsurl";
declare module "query-string";
