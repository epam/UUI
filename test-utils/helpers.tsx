import React, { ComponentType, ReactElement, ReactNode } from 'react';
import { mount } from 'enzyme';
import { renderHook } from '@testing-library/react-hooks';
import { act as reactAct } from 'react-dom/test-utils';
import renderer, { act as rendererAct } from 'react-test-renderer';
import { ContextProvider, UuiContexts } from '@epam/uui-core';

export const delay = (ms: number = 1): Promise<void> =>
    new Promise(resolve => {
        setTimeout(resolve, ms);
    });

export const testSvc = {} as UuiContexts;

export async function mountHookWithContext<TProps, TResult>(hook: (props: TProps) => TResult, initialProps?: TProps) {
    const wrapper = ({ children }: { children?: React.ReactNode }) => (
        <ContextProvider onInitCompleted={svc => Object.assign(testSvc, svc)}>{children}</ContextProvider>
    );

    const { waitForNextUpdate, rerender, ...rest } = renderHook<TProps, TResult>(hook, { wrapper, initialProps: initialProps });

    await waitForNextUpdate();

    return {
        rerender: (props: TProps) => rerender({ ...props, children: undefined }),
        waitForNextUpdate,
        ...rest,
    };
}

export const mountWithContextAsync = async (children: ReactNode, enableLegacyContext = false) => {
    const wrapper = mount(
        <ContextProvider
            onInitCompleted={svc => {
                Object.assign(testSvc, svc);
            }}
            enableLegacyContext={enableLegacyContext}
        >
            {children}
        </ContextProvider>
    );

    await reactAct(delay);
    wrapper.update();

    return wrapper;
};

export const mountWrappedComponentAsync = async <TComponent extends ComponentType<TProps>, TProps>(
    Component: TComponent,
    props: TProps,
    enableLegacyContext = false
) => {
    const renderComponent = (properties: TProps) =>
        mount(
            React.createElement(
                (p: any) => (
                    <ContextProvider
                        onInitCompleted={svc => {
                            Object.assign(testSvc, svc);
                        }}
                        enableLegacyContext={enableLegacyContext}
                    >
                        <Component {...p} />
                    </ContextProvider>
                ),
                properties
            )
        );
    const wrapper = renderComponent(props);

    await reactAct(delay);
    wrapper.update();

    return wrapper;
};

export const renderWithContextAsync = async (component: ReactElement, enableLegacyContext = false) => {
    const result = renderer.create(
        <ContextProvider
            onInitCompleted={svc => {
                Object.assign(testSvc, svc);
            }}
            enableLegacyContext={enableLegacyContext}
        >
            {component}
        </ContextProvider>
    );

    await rendererAct(delay);
    return result.toJSON();
};
//
// global = {
//     ...global,
//     delay,
//     testSvc,
//     mountWithContextAsync,
//     mountWrappedComponentAsync,
//     renderWithContextAsync,
// } as any;

// // @ts-ignore
// global.delay = delay;
// // @ts-ignore
// global.testSvc = testSvc;
// // @ts-ignore
// global.mountWithContextAsync = mountWithContextAsync;
// // @ts-ignore
// global.mountWrappedComponentAsync = mountWrappedComponentAsync;
// // @ts-ignore
// global.renderWithContextAsync = renderWithContextAsync;
