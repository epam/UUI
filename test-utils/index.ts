export * from './helpers';
export * from './mocks';
