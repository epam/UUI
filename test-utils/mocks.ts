export const windowMock = {
    matchMedia: () => ({matches: false}),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    document: {
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
    },
} as any;