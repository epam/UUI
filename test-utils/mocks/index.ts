export * from './window';
export * from './router';
