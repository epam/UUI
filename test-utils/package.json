{
	"name": "@epam/test-utils",
	"version": "4.10.2",
  "author": "EPAM",
  "license": "MIT",
  "private": true,
  "scripts": {
    "build": "node ../uui-build/scripts/buildModule"
  },
  "dependencies": {
    "@types/react-dom": "18.0.10",
    "@types/react-test-renderer": "18.0.0",
    "@epam/uui-core": "4.10.2",
    "react-test-renderer": "18.2.0",
    "@testing-library/react": "14.0.0"
	},
  "peerDependencies": {
    "react": ">=16.0.0",
    "react-dom": ">=16.0.0"
  }
}
