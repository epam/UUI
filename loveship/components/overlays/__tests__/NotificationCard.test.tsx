import React from 'react';
import { renderWithContextAsync } from '@epam/test-utils';
import {
    ClearNotification,
    ErrorNotification,
    HintNotification,
    NotificationCard,
    SuccessNotification,
    WarningNotification,
} from '../NotificationCard';
import { ReactComponent as AcceptIcon } from '../../icons/accept-12.svg';

describe('NotificationCard', () => {
    it('should be rendered correctly', async () => {
        const tree = await renderWithContextAsync(
            <NotificationCard icon={AcceptIcon} id={1} key="test" color="sun" onClose={jest.fn()} onSuccess={jest.fn()} />
        );
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered correctly with icon', async () => {
        const tree = await renderWithContextAsync(
            <NotificationCard icon={AcceptIcon} id={1} key="test" color="sun" onClose={jest.fn()} onSuccess={jest.fn()} />
        );
        expect(tree).toMatchSnapshot();
    });
});

describe('WarningNotification', () => {
    it('should be rendered correctly', async () => {
        const tree = await renderWithContextAsync(<WarningNotification id={1} key="test" onClose={jest.fn()} onSuccess={jest.fn()} />);
        expect(tree).toMatchSnapshot();
    });
});

describe('SuccessNotification', () => {
    it('should be rendered correctly', async () => {
        const tree = await renderWithContextAsync(<SuccessNotification id={1} key="test" onClose={jest.fn()} onSuccess={jest.fn()} />);
        expect(tree).toMatchSnapshot();
    });
});

describe('HintNotification', () => {
    it('should be rendered correctly', async () => {
        const tree = await renderWithContextAsync(<HintNotification id={1} key="test" onClose={jest.fn()} onSuccess={jest.fn()} />);
        expect(tree).toMatchSnapshot();
    });
});

describe('ErrorNotification', () => {
    it('should be rendered correctly', async () => {
        const tree = await renderWithContextAsync(<ErrorNotification id={1} key="test" onClose={jest.fn()} onSuccess={jest.fn()} />);
        expect(tree).toMatchSnapshot();
    });
});

describe('ClearNotification', () => {
    it('should be rendered correctly', async () => {
        const tree = await renderWithContextAsync(<ClearNotification key="test" />);
        expect(tree).toMatchSnapshot();
    });
});
