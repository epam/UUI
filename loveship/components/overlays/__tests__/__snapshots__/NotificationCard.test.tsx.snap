// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClearNotification should be rendered correctly 1`] = `
<div
  className="notificationWrapper clearButton"
>
  <button
    className="container uui-button-box uui-enabled -clickable root size-36 color-sky font-sans-semibold uui-no-left-icon uui-no-right-icon"
    onClick={[Function]}
    tabIndex={0}
    type="button"
  >
    <div
      className="uui-caption line-height-18 font-size-14 v-padding-9"
    >
      CLOSE ALL NOTIFICATIONS
    </div>
  </button>
</div>
`;

exports[`ErrorNotification should be rendered correctly 1`] = `
<div
  className="notificationWrapper color-fire typeError"
  role="alert"
>
  <div
    className="mainPath"
  >
    <div
      className="iconWrapper"
    >
      <div
        className="container uui-icon uui-enabled actionIcon"
        style={Object {}}
      >
        <svg
          className=""
        />
      </div>
    </div>
    <div
      className="content"
    />
    <div
      className="closeWrapper"
    >
      <button
        className="container uui-button-box uui-enabled -clickable icon-button-night600 root uui-theme-loveship icon-button-color-night600 closeIcon"
        onClick={[Function]}
        tabIndex={0}
        type="button"
      >
        <div
          className="container uui-icon uui-enabled"
          style={Object {}}
        >
          <svg
            className=""
          />
        </div>
      </button>
    </div>
  </div>
</div>
`;

exports[`HintNotification should be rendered correctly 1`] = `
<div
  className="notificationWrapper color-sky typeHint"
  role="alert"
>
  <div
    className="mainPath"
  >
    <div
      className="iconWrapper"
    >
      <div
        className="container uui-icon uui-enabled actionIcon"
        style={Object {}}
      >
        <svg
          className=""
        />
      </div>
    </div>
    <div
      className="content"
    />
    <div
      className="closeWrapper"
    >
      <button
        className="container uui-button-box uui-enabled -clickable icon-button-night600 root uui-theme-loveship icon-button-color-night600 closeIcon"
        onClick={[Function]}
        tabIndex={0}
        type="button"
      >
        <div
          className="container uui-icon uui-enabled"
          style={Object {}}
        >
          <svg
            className=""
          />
        </div>
      </button>
    </div>
  </div>
</div>
`;

exports[`NotificationCard should be rendered correctly 1`] = `
<div
  className="notificationWrapper color-sun"
  role="alert"
>
  <div
    className="mainPath"
  >
    <div
      className="iconWrapper"
    >
      <div
        className="container uui-icon uui-enabled actionIcon"
        style={Object {}}
      >
        <svg
          className=""
        />
      </div>
    </div>
    <div
      className="content"
    />
    <div
      className="closeWrapper"
    >
      <button
        className="container uui-button-box uui-enabled -clickable icon-button-night600 root uui-theme-loveship icon-button-color-night600 closeIcon"
        onClick={[Function]}
        tabIndex={0}
        type="button"
      >
        <div
          className="container uui-icon uui-enabled"
          style={Object {}}
        >
          <svg
            className=""
          />
        </div>
      </button>
    </div>
  </div>
</div>
`;

exports[`NotificationCard should be rendered correctly with icon 1`] = `
<div
  className="notificationWrapper color-sun"
  role="alert"
>
  <div
    className="mainPath"
  >
    <div
      className="iconWrapper"
    >
      <div
        className="container uui-icon uui-enabled actionIcon"
        style={Object {}}
      >
        <svg
          className=""
        />
      </div>
    </div>
    <div
      className="content"
    />
    <div
      className="closeWrapper"
    >
      <button
        className="container uui-button-box uui-enabled -clickable icon-button-night600 root uui-theme-loveship icon-button-color-night600 closeIcon"
        onClick={[Function]}
        tabIndex={0}
        type="button"
      >
        <div
          className="container uui-icon uui-enabled"
          style={Object {}}
        >
          <svg
            className=""
          />
        </div>
      </button>
    </div>
  </div>
</div>
`;

exports[`SuccessNotification should be rendered correctly 1`] = `
<div
  className="notificationWrapper color-grass typeSuccess"
  role="alert"
>
  <div
    className="mainPath"
  >
    <div
      className="iconWrapper"
    >
      <div
        className="container uui-icon uui-enabled actionIcon"
        style={Object {}}
      >
        <svg
          className=""
        />
      </div>
    </div>
    <div
      className="content"
    />
    <div
      className="closeWrapper"
    >
      <button
        className="container uui-button-box uui-enabled -clickable icon-button-night600 root uui-theme-loveship icon-button-color-night600 closeIcon"
        onClick={[Function]}
        tabIndex={0}
        type="button"
      >
        <div
          className="container uui-icon uui-enabled"
          style={Object {}}
        >
          <svg
            className=""
          />
        </div>
      </button>
    </div>
  </div>
</div>
`;

exports[`WarningNotification should be rendered correctly 1`] = `
<div
  className="notificationWrapper color-sun typeWarning"
  role="alert"
>
  <div
    className="mainPath"
  >
    <div
      className="iconWrapper"
    >
      <div
        className="container uui-icon uui-enabled actionIcon"
        style={Object {}}
      >
        <svg
          className=""
        />
      </div>
    </div>
    <div
      className="content"
    />
    <div
      className="closeWrapper"
    >
      <button
        className="container uui-button-box uui-enabled -clickable icon-button-night600 root uui-theme-loveship icon-button-color-night600 closeIcon"
        onClick={[Function]}
        tabIndex={0}
        type="button"
      >
        <div
          className="container uui-icon uui-enabled"
          style={Object {}}
        >
          <svg
            className=""
          />
        </div>
      </button>
    </div>
  </div>
</div>
`;
